### PR TITLE
Add table aws_eks_addon. Closes #468

### LINKED ISSUE #468: Add table aws_eks_addons
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by visiit**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT 300

SETUP: tests/aws_eks_addon []

PRETEST: tests/aws_eks_addon

TEST: tests/aws_eks_addon
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_partition.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_iam_role.named_test_resource: Creating...
aws_vpc.named_test_resource: Creating...
aws_iam_role.named_test_resource: Creation complete after 4s [id=turbottest83046]
aws_iam_role_policy_attachment.named_test_resource: Creating...
aws_iam_role_policy_attachment.named_test_resource2: Creating...
aws_iam_role_policy_attachment.named_test_resource: Creation complete after 3s [id=turbottest83046-20210610062431537200000001]
aws_iam_role_policy_attachment.named_test_resource2: Creation complete after 3s [id=turbottest83046-20210610062431538100000002]
aws_vpc.named_test_resource: Still creating... [10s elapsed]
aws_vpc.named_test_resource: Creation complete after 12s [id=vpc-0e00488e12c096cfe]
aws_subnet.named_test_resource2: Creating...
aws_subnet.named_test_resource1: Creating...
aws_subnet.named_test_resource2: Creation complete after 4s [id=subnet-0ac808f3c5c81832a]
aws_subnet.named_test_resource1: Creation complete after 4s [id=subnet-064db175b33b1a832]
aws_eks_cluster.named_test_resource: Creating...
aws_eks_cluster.named_test_resource: Still creating... [10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m20s elapsed]
aws_eks_cluster.named_test_resource: Creation complete after 10m23s [id=turbottest83046]
aws_eks_addon.named_test_resource: Creating...
aws_eks_addon.named_test_resource: Still creating... [10s elapsed]
aws_eks_addon.named_test_resource: Still creating... [20s elapsed]
aws_eks_addon.named_test_resource: Still creating... [30s elapsed]
aws_eks_addon.named_test_resource: Creation complete after 35s [id=turbottest83046:vpc-cni]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 8 added, 0 changed, 0 destroyed.

Outputs:

account_id = 123456789012
resource_aka = arn:aws:eks:us-east-1:123456789012:addon/turbottest83046/vpc-cni/36bcfa50-d6b1-339f-878e-9119703aaa93
resource_name = turbottest83046

Running SQL query: test-get-query.sql
[
  {
    "addon_name": "vpc-cni",
    "addon_version": "v1.7.5-eksbuild.2",
    "arn": "arn:aws:eks:us-east-1:123456789012:addon/turbottest83046/vpc-cni/36bcfa50-d6b1-339f-878e-9119703aaa93",
    "status": "ACTIVE"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "addon_name": "vpc-cni",
    "arn": "arn:aws:eks:us-east-1:123456789012:addon/turbottest83046/vpc-cni/36bcfa50-d6b1-339f-878e-9119703aaa93",
    "cluster_name": "turbottest83046",
    "service_account_role_arn": null
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "123456789012",
    "akas": [
      "arn:aws:eks:us-east-1:123456789012:addon/turbottest83046/vpc-cni/36bcfa50-d6b1-339f-878e-9119703aaa93"
    ],
    "tags": {},
    "title": "vpc-cni"
  }
]
✔ PASSED

POSTTEST: tests/aws_eks_addon

TEARDOWN: tests/aws_eks_addon

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
- Basic info

select
  addon_name,
  arn,
  addon_version,
  cluster_name,
  status,
  service_account_role_arn
from
  aws_eks_addon;

+------------+------------------------------------------------------------------------------------------------------------+--------------------+------------------+----------+--------------------------+
| addon_name |  arn                                                                                                       | addon_version      | cluster_name     | status   | service_account_role_arn |
+------------+------------------------------------------------------------------------------------------------------------+--------------------+------------------+----------+--------------------------+
| vpc-cni    | arn:aws:eks:ap-south-1:986325076345:addon/turbot-steampipe/vpc-cni/54bce861-5c5a-b579-591a-f5b0755cbb65    | v1.7.5-eksbuild.2  | turbot-steampipe | ACTIVE   | <null>                   |
| coredns    | arn:aws:eks:ap-south-1:986325076345:addon/turbot-steampipe/coredns/a8bce861-5c57-69a3-7096-b2b1b3899ab7    | v1.8.0-eksbuild.1  | turbot-steampipe | DEGRADED | <null>                   |
| kube-proxy | arn:aws:eks:ap-south-1:986325076345:addon/turbot-steampipe/kube-proxy/e0bce861-5c57-32de-1238-5aa42213bd28 | v1.19.6-eksbuild.2 | turbot-steampipe | ACTIVE   | <null>                   |
+------------+------------------------------------------------------------------------------------------------------------+--------------------+------------------+----------+--------------------------+

- List addons that are not active
select
  addon_name,
  arn,
  cluster_name,
  status
from
  aws_eks_addon
where
  status <> 'ACTIVE';

+------------+---------------------------------------------------------------------------------------------------------+------------------+----------+
| addon_name | arn                                                                                                     | cluster_name     | status   |
+------------+---------------------------------------------------------------------------------------------------------+------------------+----------+
| coredns    | arn:aws:eks:ap-south-1:986325076345:addon/turbot-steampipe/coredns/a8bce861-5c57-69a3-7096-b2b1b3899ab7 | turbot-steampipe | DEGRADED |
+------------+---------------------------------------------------------------------------------------------------------+------------------+----------+

- Get addon count by cluster name

select
  cluster_name,
  count(addon_name) as addon_count
from
  aws_eks_addon
group by
  cluster_name;

+------------------+-------------+
| cluster_name     | addon_count |
+------------------+-------------+
| turbot-steampipe | 3           |
+------------------+-------------+
```
</details>

### CODE REVIEW

File: aws/table_aws_eks_addon.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+				Hydrate:     getEksAddon,
+			},
+			{
+				Name:        "tags",
+				Description: "The metadata that you apply to the cluster to assist with categorization and organization.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksAddon,
+			},
+
+			// Standard columns for all tables
```
Comment (khushboo9024) [inline aws/table_aws_eks_addon.go:89]:
```suggestion
			// Steampipe standard columns
```

File: aws/table_aws_eks_addon.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "addon_name",
+				Description: "The name of the add-on.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "addon_arn",
```
Comment (khushboo9024) [inline aws/table_aws_eks_addon.go:36]: `addon_arn` should be `arn`, please change accordingly.

File: aws/table_aws_eks_addon.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+				Description: "The Amazon Resource Name (ARN) of the IAM role that is bound to the Kubernetes service account used by the add-on.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksAddon,
+			},
+			{
+				Name:        "status",
+				Description: "The status of the add-on.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksAddon,
+			},
```
Comment (Subhajit97) [inline aws/table_aws_eks_addon.go:75]: Can we add this column after `addon_version`?

File: aws/table_aws_eks_addon.go [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+				Description: "An object that represents the health of the add-on.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksAddon,
+			},
+			{
+				Name:        "tags",
+				Description: "The metadata that you apply to the cluster to assist with categorization and organization.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksAddon,
+			},
```
Comment (Subhajit97) [inline aws/table_aws_eks_addon.go:87]: Move it under standard columns

File: aws/table_aws_eks_addon.go [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+			// Standard columns for all tables
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("AddonName"),
+				Hydrate:     getEksAddon,
```
Comment (Subhajit97) [inline aws/table_aws_eks_addon.go:95]: Do we need this, since we are not using hydrate in `addon_name`?

File: aws/table_aws_eks_addon.go [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+func getEksAddon(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getEksAddon")
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	var clusterName string
+	var addonName string
```
Comment (Subhajit97) [inline aws/table_aws_eks_addon.go:151]:
```suggestion
	var clusterName, addonName string
```

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by khushboo9024 — APPROVED** [review]

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Pull request merged**